cmake/checks/check_02_compiler_features.cmake

```diff
@@ -290,21 +290,19 @@ ENDIF()
 # - Wolfgang Bangerth, 2012
 #
 
-# some compilers compile the attributes but they do not work:
+# some compilers compile the attributes but they do not work
+# so we treat errors as warnings:
```

Comment:
I don't think I understand this sentence. Are you saying the compiler "accepts the attributes when parsing the code, but then does not interpret them accordingly"?

Reply:
Some compilers compile the code happily but emit a warning (like one of the ones below) saying that they don't support the attribute properly. Do you want me to rephrase the sentence?

Reply:
Yes, maybe. I think what tripped me up was that you say "compile", but that's not the point. Compilation is the conversion into machine code, but the specific phase that's relevant is the parsing and semantic analysis: they *compile* the file just fine, they just don't *understand* correctly.

Reply:
> Compilation is the conversion into machine code, but the specific phase that's relevant is the parsing and semantic analysis
I think it is acceptable to call the process the "compiler" does is "compiling" even though it consists of several steps. That said, rephrasing this sentence won't hurt.